bin/live/pycbc_live_single_trigger_fits

```diff
@@ -273,30 +285,98 @@ logging.info('Sorting events into template duration bins')
 n_bins = duration_bin_edges.size - 1
 alphas = {i: np.zeros(n_bins, dtype=np.float32) for i in args.ifos}
 counts = {i: np.zeros(n_bins, dtype=np.float32) for i in args.ifos}
-binned_sngl_stats = {ifo: {} for ifo in args.ifos}
+event_bins = {}
+times_to_prune = {ifo: [] for ifo in args.ifos}
+
+# Note: turn off logging for cluster_over_time which is quite verbose
+logger = logging.getLogger()
+# Remember current logging level
+level = logger.level
+logger.setLevel(logging.WARN)
 
 for ifo in events.keys():
     # Sort the events into their bins
-    event_bin = np.array([tbins[d]
-                          for d in events[ifo].data['template_duration']])
+    event_bins[ifo] = np.array([tbins[d]
+                               for d in events[ifo].data['template_duration']])
+
+    if args.prune_loudest:
+        for bin_num in range(n_bins):
+            inbin = event_bins[ifo] == bin_num
+
+            binned_events = events[ifo].data[args.sngl_ranking][inbin]
+            binned_event_times = events[ifo].data['end_time'][inbin]
+
+            # Cluster the triggers in time (use the pruning window)
+            # to make sure we aren't just getting the same event N times
+            # in different templates within the same bin
+            cidx = cluster_over_time(binned_events, binned_event_times,
+                                     args.prune_window)
+
+            # Of these events, are any of them above the specified
+            # minimum statistic value?
+            above_stat_min = binned_events[cidx] >= args.minimum_prune_stat
+            cidx = cidx[above_stat_min]
+
+            if args.prune_loudest > cidx.size:
+                # There are fewer clusters than the number specified,
+                # so prune them all
+                times_to_prune[ifo] += list(binned_event_times[cidx])
+                continue
+
+            # Find the loudest of the triggers in this bin
+            argloudest = np.argsort(binned_events[cidx])[-args.prune_loudest:]
+            times_to_prune[ifo] += list(binned_event_times[cidx][argloudest])
+
+# Turn logging back on
+logger.setLevel(level)
+
+n_pruned = {ifo: [] for ifo in args.ifos}
+pruned_trigger_times = {}
+if args.prune_loudest:
+    logging.info("Pruning triggers in a %.2fs window around the loudest %d "
+                 "triggers in each bin", args.prune_window,
+                 args.prune_loudest)
+    for ifo in args.ifos:
+        times = events[ifo].data['end_time'][:]
+        outwith_window = np.ones_like(times, dtype=bool)
+        for t in times_to_prune[ifo]:
+            outwith_window &= abs(times - t) > args.prune_window
+            # Need to make an (ever-so-small) correction to the live time
+            live_time[ifo] -= args.prune_window
```

Comment:
should this now be 2 * args.prune_window ?

Reply:
I've updated comments and implementation so args.prune_window is correct now

Reply:
yes, but how much live time is actually being lost?

Reply:
Ah, good point - sorry I missed the timing of the comment. Updated now